src/System.Private.ServiceModel/src/System/ServiceModel/Channels/Connection.cs

```diff
@@ -31,6 +31,62 @@ public interface IConnection
         int EndRead();
     }
 
+    public static class IConnectionExtensions
+    {
+        public static async Task WriteAsync(this IConnection connection, byte[] buffer, int offset, int size, bool immediate, TimeSpan timeout)
```

Comment:
Added comments on these methods. Any suggestions on where we want to place these methods?